erasure-utils.go

```diff
@@ -235,3 +236,48 @@ func copyBuffer(writer io.Writer, disk StorageAPI, volume string, path string, b
 	// Success.
 	return nil
 }
+
+// Error returned when there are no more free buffers in the scratch pool.
+// Normally this error should never be returned as scratch pool should always be
```

Comment:
I am not entirely comfortable with this approach, it creates complicated code base. 

We should use 'sync.Pool' if necessary but previously using that never really helped entirely.  

Code as below is not well tested and i would avoid adding these type of primitives in the codebase for simplicity sake.